pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/PinotFSSegmentUploader.java

```diff
@@ -74,7 +73,7 @@ public URI uploadSegment(File segmentFile, LLCSegmentName segmentName, int timeo
     final String rawTableName = TableNameBuilder.extractRawTableName(segmentName.getTableName());
     Callable<URI> uploadTask = () -> {
       URI destUri = new URI(StringUtil.join(File.separator, _segmentStoreUriStr, segmentName.getTableName(),
-          segmentName.getSegmentName() + UUID.randomUUID().toString()));
+          SegmentCompletionUtils.generateSegmentFileName(segmentName.getSegmentName())));
```

Comment:
Server2ControllerSegmentUploader and PinotFSSegmentUploader has different naming pattern for tmp segments before.
Server2ControllerSegmentUploader: segment_name.tmp.<UUID>
PinotFSSegmentUploader: segment_name.<UUID>